processor/metricstransformprocessor/datapoint_aggregation.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// timeseriesGroupByLabelValues is a data structure for grouping timeseries that will be aggregated
+type timeseriesGroupByLabelValues struct {
+	timeseries  []*metricspb.TimeSeries
+	labelValues []*metricspb.LabelValue
+}
+
+// aggregateTimeseriesGroups attempts to merge each group of timeseries in keyToTimeseriesMap through the specific calculation indicated by aggrType and dataType to construct the new timeseries with new label values indicated in keyToLabelValuesMap
+// Returns the new list of timeseries
+func (mtp *metricsTransformProcessor) aggregateTimeseriesGroups(keyToTimeseriesMap map[string]*timeseriesGroupByLabelValues, aggrType AggregationType, dataType metricspb.MetricDescriptor_Type) []*metricspb.TimeSeries {
```

Comment:
For that SDK, does that "merge" and the "aggregate" here imply the same thing? If not, having the same name might cause confusions later.

Reply:
Iiuc, it's pretty much the same concept. In the SDK, an `aggregator` defines an aggregation of a metric instrument, e.g. a "Sum" of individual measurements (note in the processor we are never dealing with individual measurements, a single data point already represents an aggregation by the SDK). The SDK `Merge` function merges two aggregators, i.e. if the aggregator type was "Sum" it will add the sum values from the two aggregators together.

To me, that seems the same as what we are doing here: merging two data points that each represent aggregations, using a defined aggregation type.

e.g. for Sum aggregation: https://github.com/open-telemetry/opentelemetry-go/blob/2966505271c3698127db3d243ff86682e22caff0/sdk/metric/aggregator/sum/sum.go#L78

Reply:
perfect! Thanks.